system-test/speech.js

```diff
@@ -0,0 +1,411 @@
+  };
+});
+
+describe('Speech', function() {
+  var speech;
+  var bucket = storage.bucket(BUCKET_NAME);
+
+  before(function(done) {
+    async.waterfall([
+      function(next) {
+        bucket.create(next);
+      },
+      function(zone, apiResponse, next) {
+        async.map(FILENAMES, function(filename, onComplete) {
+          fs.readFile(AUDIO_FILES[filename].path, onComplete);
+        }, next);
+      },
+      function(files, next) {
+        FILENAMES.forEach(function(filename, i) {
+          AUDIO_FILES[filename].content = files[i];
+        });
+        async.map(FILENAMES, function(filename, onComplete) {
+          var file = bucket.file(AUDIO_FILES[filename].name);
+          file.save(AUDIO_FILES[filename].content, function(err) {
+            onComplete(err, file);
+          });
+        }, next);
+      },
+      function(files, next) {
+        async.map(files, function(file, onComplete) {
+          file.makePublic(onComplete);
+        }, next);
+      }
+    ], done);
+  });
+
+  beforeEach(function() {
+    speech = gcloud.speech();
+  });
+
+  after(function(done) {
+    bucket.getFiles(function(err, files) {
+      if (err) {
+        done(err);
+        return;
+      }
+
+      async.map(files, function(file, onComplete) {
+        file.delete(onComplete);
+      }, function(err) {
+        if (err) {
+          done(err);
+          return;
+        }
+
+        bucket.delete(done);
+      });
+    });
+  });
+
+  describe('recognize', function() {
+    it('recognizes speech from raw audio', function(done) {
+      fs.readFile(AUDIO_FILES.bridge.path, function(err, audioFile) {
+        assert.ifError(err);
+        speech.recognize({
+          config: {
+            encoding: 'LINEAR16',
+            sampleRate: 16000
+          },
+          audio: {
+            content: audioFile
+          }
+        }, function(err, response, apiResponse) {
+          assert.ifError(err);
+          assert(Array.isArray(response.results));
+          assert(response.results.length > 0);
+          assert(apiResponse);
+          assert(apiResponse.results);
+          done();
+        });
+      });
+    });
+    it('recognizes speech from remote GCS audio file', function(done) {
+      speech.recognize({
+        config: {
+          encoding: 'LINEAR16',
+          sampleRate: 16000
+        },
+        audio: {
+          uri: AUDIO_FILES.bridge.gcsUri
+        }
+      }, function(err, response, apiResponse) {
+        assert.ifError(err);
+        assert(Array.isArray(response.results));
+        assert(response.results.length > 0);
+        assert(apiResponse);
+        assert(apiResponse.results);
+        done();
+      });
+    });
+  });
+
+  describe('detect', function() {
+    it('recognizes speech from raw audio', function(done) {
+      speech.detect(AUDIO_FILES.bridge.path, {
+        encoding: 'LINEAR16',
+        sampleRate: 16000
+      }, function(err, response, apiResponse) {
```

Comment:
Technically it's an array of speech result alternatives, I just got lazy and named the variable `response`. I'll rename.